File: 2021/days/d1.rb
require "parallel"

module AOC
    class Day1 < AocDay
        def countConsecutiveIncreases(e)
            e.each_with_index.map { |m, i| 
                0 if i == 0
                m > e.map(&:to_i)[i-1] ? 1 : 0
            }.reduce(&:+)
        end

        def part1
            @p1 = countConsecutiveIncreases @lines.map(&:to_i)
        end

        def part2
            @p2 = countConsecutiveIncreases @lines.map(&:to_i).each_cons(3).map{|e|
                e.reduce(&:+)
            }
        end
    end
end

File: 2019/days/d1.cs
namespace AOC {
    class Day1 : AoCDay {
        public Day1(string linesRaw): base(linesRaw) {}
        
        override public string Part1() {
            return this.lines.Select(m => {
                var i = Int32.Parse(m);
                return i / 3 -2;
            }).Sum().ToString();
        }

        override public string Part2() {
            return this.lines.Select(m => {
                var i = Int32.Parse(m);
                var masses = new List<Int32>();
                masses.Add(i / 3 - 2);
                do {
                    masses.Add(masses.Last() / 3 - 2);
                } while (masses.Last() > 0);

                return masses.GetRange(0, masses.Count()-1).Sum();
            }).Sum().ToString();
        }
    }
}